### PR TITLE
feat: implement `Timestamp` object

### LINKED ISSUE #16: request: `Timestamp` object
### What's the feature?

A `Timestamp` object would allow us to imitate what `datetime.datetime` already provides to us, but with `created_at` and `joined_at` fields (if applicable) for resource dataclasses that may not supply it.

### How does it work?

```py
from attrs import define, field
from datetime import datetime
from enum import Enum

# This import stands relative to where we declare.
from ...utils.converters import optional_c
from ...const import NotNeeded, MISSING

class TimestampStyle(Enum):
    SHORT_TIME = "t"
    LONG_TIME = "T"
    SHORT_DATE = "d"
    LONG_DATE = "D"
    SHORT_DATE_TIME = "f"
    LONG_DATE_TIME = "F"
    RELATIVE_TIME = "R"

@define(repr=False, eq=False)
class Timestamp:
    _timestamp: str | datetime = field(converter=datetime.fromisoformat)
    created_at: str | datetime | None = field(converter=optional_c(datetime.fromisoformat), default=None)
    joined_at: str | datetime | None = field(converter=optional_c(datetime.fromisoformat), default=None)

    def __repr__(self) -> datetime:
        return self._timestamp

    def __eq__(self, other: str | datetime) -> bool:
        if type(other) == str:
            return bool(str(self._timestamp) == other)
        else:
            return bool(self._timestamp == other)

    def mention(self, style: NotNeeded[str | TimestampStyle] = MISSING) -> str:
        if style is MISSING:
            _style = "f"
        else:
            _style = style if isinstance(style, str) else style.value

        return f"<t:{self._timestamp}:{_style}>"
```

### Code of Conduct

- [X] I agree.

### PR DESCRIPTION
**Pull request by i0bs**
This implements the `Timestamp` object and resolves #16 .

### CODE REVIEW

File: retux/client/resources/abc.py [hunk 1]
```diff
@@ -396,7 +402,81 @@ def path(self) -> str:
...
+    SHORT_DATE_TIME = "f"
+    """Formats the timestamp as DD Jj YYYY HH:mm."""
+    LONG_DATE_TIME = "F"
+    """Formats the timestamp as J, DD Jj YYYY HH:mm."""
+    RELATIVE = "R"
+    """Formats the timestamp as `x` (DD/MM/YY)."""
+
+
+@define(repr=False, eq=False)
+class Timestamp:
```
Comment (Dworv) [inline retux/client/resources/abc.py:446]: Is there a reason that you don't have a docstring here?

**Review by Dworv — CHANGES_REQUESTED** [review]
Seems to me like `Timestamp` is missing a class docstring.

File: retux/client/resources/abc.py (hunk 1 above)
Comment (i0bs) [reply retux/client/resources/abc.py:446]: I forgor 💀

**Review by Dworv — APPROVED** [review]
LGTM

**Pull request merged**